pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -218,12 +219,21 @@ func filterSeedsMatchingSeedSelector(cloudProfile *gardencorev1beta1.CloudProfil
 	return matchingSeeds, nil
 }
 
-func getCandidates(shoot *gardencorev1beta1.Shoot, seedList []*gardencorev1beta1.Seed, strategy config.CandidateDeterminationStrategy) ([]*gardencorev1beta1.Seed, error) {
+func getCandidates(cloudProfile *gardencorev1beta1.CloudProfile, shoot *gardencorev1beta1.Shoot, seedList []*gardencorev1beta1.Seed, strategy config.CandidateDeterminationStrategy) ([]*gardencorev1beta1.Seed, error) {
 	var candidates []*gardencorev1beta1.Seed
+	var providers []string
 
+	if cloudProfile.Spec.SeedSelector != nil {
+		providers = cloudProfile.Spec.SeedSelector.Providers
+	}
 	switch {
 	case shoot.Spec.Purpose != nil && *shoot.Spec.Purpose == gardencorev1beta1.ShootPurposeTesting:
 		candidates = determineCandidatesOfSameProvider(seedList, shoot)
+	case strategy == config.Selector:
+		if matchProvider(cloudProfile.Spec.Type, shoot.Spec.Provider.Type, providers) {
```

Comment:
The signature of the `matchProvider` function suggests that the first argument is the seed's provider, but you are passing the provider of the `CloudProfile`. I'm confused.

Also, why are we executing this function here? We are calling it again inside `addCanditatesWithMatchingProviders`?!

Reply:
This was to always start the list of candidates with matching region seeds if the shoots provider is in the list of accepted providers. This should prefer those seeds

Reply:
Hm, can it be made more readable? It's not too straight-forward to understand if you just read these lines.

Reply:
This strategy has been deleted completely now.